timemachine/cpp/src/segmented_logsumexp.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "device_buffer.hpp"
+
+namespace timemachine {
+
+template <typename RealType> class SegmentedLogSumExp {
+
+private:
+    const int max_vals_per_segment_;
+    const int num_segments_;
+    DeviceBuffer<RealType> d_temp_buffer_;
+    // Buffers for finding the max/summed value
+    std::size_t temp_storage_bytes_;
+    DeviceBuffer<char> d_temp_storage_buffer_;
+
+public:
+    SegmentedLogSumExp(const int max_vals_per_segment, const int num_segments);
+
+    ~SegmentedLogSumExp();
+
+    void sum_device(
+        const int total_values,
+        const int num_segments,
+        const int *d_segment_offsets,
+        const RealType *d_values,
+        RealType *d_max_sum_out,
```

Comment:
`d_max_sum_out` is written `d_max_out` in the `.cu` files.

Reply:
Fixed in https://github.com/proteneer/timemachine/pull/1239/commits/079443277fba0a34f1fcb030d14c7f7b5951d09c